lambda/js/src/main/scala/feral/lambda/IOLambdaPlatform.scala

```diff
@@ -34,10 +36,24 @@ private[lambda] trait IOLambdaPlatform[Event, Result] {
   private lazy val handlerFn
       : js.Function2[js.Any, facade.Context, js.Promise[js.Any | Unit]] = {
     (event: js.Any, context: facade.Context) =>
-      (for {
-        lambda <- setupMemo
-        event <- IO.fromEither(decodeJs[Event](event))
-        result <- lambda(event, Context.fromJS(context))
-      } yield result.map(_.asJsAny).orUndefined).unsafeToPromise()(runtime)
+      {
+        // how to extract?
+        // need to combinate with below?
+        setupMemo.toJSPromise(scala.concurrent.ExecutionContext.global)
```

Comment:
So the same way you can combine `Future` with `flatMap`, you can combine `js.Promise` with `then`.

Also, you can use `runtime.compute` instead of `ExecutionContext.global`.